priv/repo/migrations/20160803181844_replace_username_with_slack_handle.exs

```diff
@@ -0,0 +1,35 @@
+defmodule Cog.Repo.Migrations.ReplaceUsernameWithSlackHandle do
+  use Ecto.Migration
+  alias Cog.Repo
+  alias Cog.Models.User
+
+  def change do
+    users = User
+    |> Repo.all
+    |> Repo.preload([chat_handles: :chat_provider])
```

Comment:
In previous migrations when we've used models directly, it has bitten us badly, causing us to have to go back and rewrite old migrations. Using model code ends up coupling the database migration to the state of Cog code _when the migration was written_; migrations should only be coupled to the state of the database resulting from the previous migration. If we change the structure of the models in the future (e.g., perhaps we no longer have a `ChatProvider` model? Depending on how the new Adapter work goes, we might not) then this could end up blowing up in the middle of a migration, perhaps making it impossible to install a new bot.

This can be rewritten to use an `execute` call with an [anonymous PL/pgsql block](https://www.postgresql.org/docs/9.0/static/sql-do.html)... I had to do something similar [here](https://github.com/operable/cog/blob/af167bc9e842b4601a0902d96f0beeae0d4f809f/priv/repo/migrations/20160505200538_bundle_version_schema.exs#L181-L203)

You might also be able to use `Cog.Repo` calls referring to raw table names and columns, instead of using models, if that makes the code easier to write. I think that should provide sufficient insulation, too.